pkg/server/proto/server.proto

```diff
@@ -392,6 +392,22 @@ service Waypoint {
   // based on the trigger configuration.
   rpc RunTrigger(RunTriggerRequest) returns (RunTriggerResponse);
 
+  //----------------------------------------------------------------------
+  // Task functions
+  //----------------------------------------------------------------------
+
+  // UpsertTask updates or inserts a Task
+  rpc UpsertTask(UpsertTaskRequest) returns (UpsertTaskResponse);
```

Comment:
I would reorganize some of these (just the location). I think `UpsertTask` should go up under the "internal metadata endpoints" section because this should never be called by any public API user.